client.py

```diff
@@ -0,0 +1,233 @@
+            responses = stub.transcode(
+                FFmpegRequest(ffmpeg_arguments=ffmpeg_arguments),
+                metadata=[('x-api-key', api_key)])
+            writer.write_command(ffmpeg_arguments, responses)
+    except KeyboardInterrupt:
+        responses.cancel()
+    writer.close()
+
+
+def _get_writer(output_format, output_file):
+    if output_format == 'json':
+        return JsonWriter(output_file)
+    elif output_format == 'tsv':
+        return TSVWriter(output_file)
+    return NormalWriter(output_file)
+
+
+def _get_ffmpeg_commands(args):
+    """Obtains all the ffmpeg commands that need to be run."""
+    if args.input_file is not None:
+        for line in args.input_file:
+            yield shlex.split(line)
+    else:
+        yield args.ffmpeg_arguments
+
+
+class ResponseWriter:
+    """Base class for all writer classes."""
+
+    def __init__(self, output_file):
+        self.output_file = output_file
+
+    def write_command(self, command, responses):
+        """Writes command and its responses."""
+
+    def close(self):
+        """Cleans up writer by closing file."""
+        self.output_file.close()
+
+
+class NormalWriter(ResponseWriter):
+    """Writes output in human-readable format."""
+
+    def write_command(self, command, responses):
+        """Writes command and its responses in human-readable format."""
+        with contextlib.redirect_stdout(self.output_file):
+            print(command)
+            for response in responses:
+                if response.HasField('log_line'):
+                    print(response.log_line, end='')
+                else:
+                    exit_code = _convert_exit_code(
+                        response.exit_status.exit_code)
+                    if exit_code >= 0:
+                        print(f'Exited with code {exit_code}')
+                    else:
+                        print(f'Killed by signal {-exit_code}')
+                    print(response.exit_status.resource_usage)
+                    print(f'Real time: {response.exit_status.real_time}')
+
+
+class JsonWriter(ResponseWriter):
+    """Writes output in JSON format.
+
+    The output is not pretty printed.
+    The JSON written is an array of objects written by write_command.
+    """
+
+    def __init__(self, output_file):
+        super().__init__(output_file)
+        self._add_comma = False
+        print('[', end='', file=self.output_file)
+
+    def write_command(self, command, responses):
+        """Writes command and its responses in JSON format.
+
+        Here is rough example of the resulting JSON:
+        {
+          "command": ["ffmpeg", "-i", "input_file", "output_file"],
+          "response": {
+            "logs": [...],
+            "exit_status": {...}
+          }
+        }
+
+        The exit_status object follows the structure of the ExitStatus protobuf.
+        """
+        with contextlib.redirect_stdout(self.output_file):
+            if self._add_comma:
```

Comment:
would be cleaner , if you construct the whole dict first, and then do json pretty print the whole dict using
json.dumps(dict, sort_keys=True, indent=4)